### PR TITLE
removing extra flush call

### PR DESCRIPTION
**Pull request by vishalchangrani**
Found an issue with the outbound network message size reporting for one to one messages. All message size of one to one message is being reported as 0

### CODE REVIEW

File: network/gossip/libp2p/middleware.go [hunk 1]
```diff
@@ -306,12 +306,6 @@ func (m *Middleware) SendDirect(msg *message.Message, targetID flow.Identifier)
 		return fmt.Errorf("failed to send message to %s: %w", targetID.String(), err)
 	}
 
-	// flush the stream
-	err = bufw.Flush()
```
Comment (vishalchangrani) [inline network/gossip/libp2p/middleware.go:310]: there is a flush at line 319. This flush cleans the buffer and the `bytecount` on line becomes 0 and consequently the outbound message size reported becomes 0.

File: network/gossip/libp2p/middleware.go [hunk 2]
```diff
@@ -306,12 +306,6 @@ func (m *Middleware) SendDirect(msg *message.Message, targetID flow.Identifier)
...
-	if err != nil {
-		return fmt.Errorf("failed to flush stream for %s: %w", targetID.String(), err)
-	}
-
 	// track the number of bytes that will be written to the wire for metrics
 	byteCount := bufw.Buffered()
```
Comment (Kay-Zee) [inline network/gossip/libp2p/middleware.go:310]: Any reason we don't just use `msg.Size()`? Are we certain that the stream doesn't auto flush?

Comment (vishalchangrani) [reply network/gossip/libp2p/middleware.go:310]: msg.Size() will actually work just fine. I will change the code to use msg.Size().

Comment (vishalchangrani) [reply network/gossip/libp2p/middleware.go:310]: I picked up the idea of flush from an libp2p example.

Comment (vishalchangrani) [reply network/gossip/libp2p/middleware.go]: ok - so the `flush` is indeed needed. If I remove it, nothing gets written to the stream failing the unit tests.

**Review by Kay-Zee — APPROVED** [review]

**Pull request merged**